scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -0,0 +1,544 @@
+colnames(pseudobulked_sce) <- paste(pseudobulked_sce$celltype_broad, 
+                                    pseudobulked_sce$sample, sep = "-")
+
+pseudobulked_sce
+```
+
+How does the new pseudo-bulked `SingleCellExperiment` look different? 
+
+```{r}
+# only 37 columns in the counts assay 
+counts(pseudobulked_sce)[1:10, 1:10]
+```
+
+Let's take a look at what the `colData` now looks like in the pseudo-bulked SCE object. 
+
+```{r}
+# note the new addition of the column with number of cells per group 
+colData(pseudobulked_sce)
+```
+
+### Filtering the pseudo-bulked `SingleCellExperiment` object
+
+*Add a bit more justification on filtering and why we need it*
+
+```{r}
+# filter to only the HVG list
+hv_genes <- metadata(integrated_sce)$combined_hvg
+filtered_pseudobulked_sce <- pseudobulked_sce[hv_genes, ]
+```
+
+We want to remove any groups in the pseudo-bulked dataset that may have a low number of cells. 
+
+```{r}
+# filter by number of cells 
+filtered_pseudobulked_sce <- filtered_pseudobulked_sce[, pseudobulked_sce$ncells >= 10]
+```
+
+We also want to filter out any genes that may be lowly expressed in our pseudo-bulked dataset.
+
+*Talk about how this specific function uses the sample groups provided to determine the minimum sample size that genes are expressed in.*
+
+```{r}
+# filter out genes that might be lowly expressed using filterbyExpr in `edgeR`
+# provide sample groups from SCE to determine minimum sample size that genes should be expressed in 
+# we are grouping samples as ERMS vs. ARMS so use subdiagnosis
+genes_filter <- edgeR::filterByExpr(filtered_pseudobulked_sce,
+                                     group = filtered_pseudobulked_sce$subdiagnosis)
+# filter by genes
+filtered_pseudobulked_sce <- filtered_pseudobulked_sce[genes_filter,]
+
+```
+
+
+```{r}
+dim(pseudobulked_sce)
+dim(filtered_pseudobulked_sce)
+```
+
+### Perform differential expression with DESeq2
+
+*Intro about DESeq2*
+
+Before we identify differentially expressed genes, we will want to subset the pseudo-bulked `SingleCellExperiment` object to contain only the cell type that we are interested in comparing across the two sub types.
+
+```{r}
+tumor_myoblast_sce <- filtered_pseudobulked_sce[, filtered_pseudobulked_sce$celltype_broad == "Tumor_Myoblast"]
+```
+
+This leaves us with 6 columns in our dataset that should all have `celltype_broad = Tumor_Myoblast`. 
+
+```{r}
+table(tumor_myoblast_sce$celltype_broad)
+```
+
+Now we are ready to set up our `DESeq2` object. 
+
+```{r}
+# deseq2 requires the coldata object as a separate input
+coldata_df <- as.data.frame(colData(tumor_myoblast_sce))
+  
+# set up the deseq object 
+deseq_object <- DESeq2::DESeqDataSetFromMatrix(counts(tumor_myoblast_sce),
+                                               colData = coldata_df,
+                                               design = ~ diagnosis_groups)
+```
+
+Use the median of ratios method for count normalization followed by regularized log transformation.
```

Comment:
Again we probably want to reiterate here that we have to normalize "again" because we summed up the raw counts and now have "new" raw counts to normalize. The previous single-cell-resolution logcounts are no longer applicable in this analysis.